### DISCUSSION TITLE
Spark API - Support for Range

### DISCUSSION THREAD

**Discussion by IndexSeek** [duckdb/duckdb#9181] [Ideas]
### What happens?

https://duckdb.org/docs/api/python/spark_api

The current `range` function exists but is not yet implemented. 

https://github.com/duckdb/duckdb/blob/511c09323f1978d11e99f25f776d37719d87ffca/tools/pythonpkg/duckdb/experimental/spark/sql/session.py#L171-L174

I thought this could be implemented using some of DuckDB's `generate_series`. 

```python
def range(start: int, end: Optional[int] = None, step: int = 1, numPartitions: Optional[int] = None) -> "DuckDBPyRelation":
    if end is None:
        raise ValueError("End value must be specified")

    query = f'SELECT * FROM generate_series({start}, {end - 1}, {step})'
    return con.sql(query)
```

I wasn't sure if this should return a DataFrame or a DuckDBPyRelation.

I think range will be useful for many people wanting to try out this new API. 

### To Reproduce

```
from duckdb.experimental.spark.sql import SparkSession as Session

session = Session.builder.getOrCreate()
session.range(1, 10).show()
```

### OS:

macOS 13.4.1 osx-arm64

### DuckDB Version:

0.9.0

### DuckDB Client:

Python

### Full Name:

Tyler White

### Affiliation:

Snowflake

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Tishj** [comment]
Hi, thanks for expressing interest in the API!

However the idea behind ContributionsAcceptedError is that it's not on the DuckDBLabs roadmap to support this (maybe later we will, but for now we don't).

As you said though, the implementation seems trivial and it can be done in pure python so the threshold to make a PR is really low, and we'll happily review contributions 👍
